android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -267,6 +267,27 @@ abstract class AbstractFetchDownloadService : Service() {
                 )
                 handleDownloadIntent(newDownloadState)
             }
+
+            ACTION_CANCEL -> {
+                val newDownloadState = download.copy(status = CANCELLED)
+                store.dispatch(
+                    DownloadAction.UpdateDownloadAction(
+                        newDownloadState,
+                    ),
+                )
+                val currentDownloadJobState = downloadJobs[download.id]
+                currentDownloadJobState?.let {
+                    it.lastNotificationUpdate = System.currentTimeMillis()
+                    setDownloadJobStatus(it, CANCELLED)
+                    it.job?.cancel()
+
+                    it.job = CoroutineScope(IO).launch {
+                        deleteDownloadingFile(it.state)
+                        it.downloadDeleted = true
+                    }
+                    removeDownloadJob(it)
+                }
```

Comment:
This looks really similar to what we are doing for [ACTION_CANCEL](https://github.com/mozilla-mobile/firefox-android/blob/5aac35da96f041c8f78358fd44e8af87cd35a46d/android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L190-L199)



I would rather prefer that we update `handleRemovePrivateDownloadIntent` which was not cancelling  the `DownloadJob`and reuse our existing functionally  for [ACTION_CANCEL](https://github.com/mozilla-mobile/firefox-android/blob/5aac35da96f041c8f78358fd44e8af87cd35a46d/android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L190-L199)
 than duplicating it.

Sharing a possible approach: 

```diff
diff --git a/android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt b/android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt
index b67a8c7bde..4f14e03e61 100644
--- a/android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt
+++ b/android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt
@@ -187,16 +187,7 @@ abstract class AbstractFetchDownloadService : Service() {
                     }
 
                     ACTION_CANCEL -> {
-                        removeNotification(context, currentDownloadJobState)
-                        currentDownloadJobState.lastNotificationUpdate = System.currentTimeMillis()
-                        setDownloadJobStatus(currentDownloadJobState, CANCELLED)
-                        currentDownloadJobState.job?.cancel()
-
-                        currentDownloadJobState.job = CoroutineScope(IO).launch {
-                            deleteDownloadingFile(currentDownloadJobState.state)
-                            currentDownloadJobState.downloadDeleted = true
-                        }
-
+                        cancelDownloadJob(context, currentDownloadJobState)
                         removeDownloadJob(currentDownloadJobState)
                         emitNotificationCancelFact()
                         logger.debug("ACTION_CANCEL for ${currentDownloadJobState.state.id}")
@@ -242,6 +233,21 @@ abstract class AbstractFetchDownloadService : Service() {
         }
     }
 
+    private fun cancelDownloadJob(
+        context: Context,
+        currentDownloadJobState: DownloadJobState,
+    ) {
+        removeNotification(context, currentDownloadJobState)
+        currentDownloadJobState.lastNotificationUpdate = System.currentTimeMillis()
+        setDownloadJobStatus(currentDownloadJobState, CANCELLED)
+        currentDownloadJobState.job?.cancel()
+
+        currentDownloadJobState.job = CoroutineScope(IO).launch {
+            deleteDownloadingFile(currentDownloadJobState.state)
+            currentDownloadJobState.downloadDeleted = true
+        }
+    }
+
     override fun onBind(intent: Intent?): IBinder? = null
 
     override fun onCreate() {
@@ -279,6 +285,7 @@ abstract class AbstractFetchDownloadService : Service() {
     internal fun handleRemovePrivateDownloadIntent(download: DownloadState) {
         if (download.private) {
             downloadJobs[download.id]?.let {
+                cancelDownloadJob(context, it)
                 removeDownloadJob(it)
             }
             store.dispatch(DownloadAction.RemoveDownloadAction(download.id))
```

Reply:
Done